specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json

```diff
@@ -13221,6 +16065,111 @@
         }
       }
     },
+    "FunctionEnvelope": {
+      "description": "Web Job Information.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "FunctionEnvelope resource specific properties",
+          "properties": {
+            "name": {
+              "description": "Function name.",
+              "type": "string"
+            },
+            "functionAppId": {
```

Comment:
read only.

Reply:
I think @ravbhatnagar  is talking about the function app id.

Reply:
The [Kudu sources](https://github.com/projectkudu/kudu/blob/master/Kudu.Contracts/Functions/FunctionEnvelope.cs) don't say that any of the FunctionEnvelope properties are read-only.  The FunctionEnvelope is passed to the CreateFunction API.  I don't know if the Kudu code expects the `functionAppId` to be set or not during the CreateFunction.

Reply:
If setting it throws an error - then it should be read only. If setting it does not throw an error - then if I set it with a random string, what does server return? If the server returns me the same random data, then the function id is wrong; if server returns me the correct function id, then it's not idempotent.

Reply:
Confirmed with owning devs: these fields (name, functionAppId) are read-only and not used when creating.  The are used when returning a collection of FunctionEnvelopes.